src/main/java/com/autotune/analyzer/serviceObjects/BulkJobStatus.java

```diff
@@ -115,130 +108,310 @@ public void setExperiments(Map<String, Experiment> experiments) {
         this.experiments = experiments;
     }
 
-    public Webhook getWebhook() {
-        return webhook;
+    public void copyByPattern(String regex) {
+
+        experiments.clear();
+        synchronized (experimentMap) {
+            if (regex == null) {
+                experiments.putAll(experimentMap);
+            } else {
+                LOGGER.debug("regex : {}", regex);
+                Pattern pattern = Pattern.compile(".*" + Pattern.quote(regex) + ".*");
+
+                experimentMap.forEach((key, value) -> {
+                    if (pattern.matcher(key).matches()) {
+                        experiments.put(key, value); // Direct reference copy
+                    }
+                });
+            }
+        }
     }
 
-    public void setWebhook(Webhook webhook) {
-        this.webhook = webhook;
-    }
+    public static enum NotificationType {
+        ERROR("error"),
+        WARNING("warning"),
+        INFO("info");
 
-    // Method to add a new experiment with "unprocessed" status and null notification
-    public synchronized Experiment addExperiment(String experimentName) {
-        Experiment experiment = new Experiment(experimentName);
-        experiments.put(experimentName, experiment);
-        return experiment;
-    }
+        private final String type;
 
-    public String getStatus() {
-        return status;
-    }
+        NotificationType(String type) {
+            this.type = type;
+        }
 
-    public void setStatus(String status) {
-        this.status = status;
+        public String getType() {
+            return type;
+        }
     }
 
-    public int getTotal_experiments() {
-        return total_experiments;
-    }
+    @JsonFilter("summaryFilter")
```

Comment:
Replace strings with constants for the `summaryFilter` , `start` and `end` time

Reply:
done